symfony/src/Entity/DeceasedReport.php

```diff
@@ -0,0 +1,315 @@
+    {
+        $this->reviewedOn = $reviewedOn;
+        return $this;
+    }
+
+    public function getDenialReason()
+    {
+        return $this->denialReason;
+    }
+
+    public function setDenialReason($denialReason)
+    {
+        $this->denialReason = $denialReason;
+        return $this;
+    }
+
+    public function getDenialReasonOtherDescription()
+    {
+        return $this->denialReasonOtherDescription;
+    }
+
+    public function setDenialReasonOtherDescription($denialReasonOtherDescription)
+    {
+        $this->denialReasonOtherDescription = $denialReasonOtherDescription;
+        return $this;
+    }
+
+    /* Model Methods */
+
+    public function getReportmechanismDisplay()
+    {
+        if (isset(self::MECHANISMS[$this->reportMechanism])) {
+            return self::MECHANISMS[$this->reportMechanism];
+        }
+        return $this->reportMechanism;
+    }
+
+    public function getReportStatusDisplay()
+    {
+        if (isset(self::STATUSES[$this->reportStatus])) {
+            return self::STATUSES[$this->reportStatus];
+        }
+        return $this->reportStatus;
+    }
+
+    public function getNextOfKinRelationshipDisplay()
+    {
+        if (isset(self::NK_RELATIONSHIPS[$this->nextOfKinRelationship])) {
+            return self::NK_RELATIONSHIPS[$this->nextOfKinRelationship];
+        }
+        return $this->nextOfKinRelationship;
+    }
+
+    public function loadFromFhirObservation($report)
+    {
+        $this->setId($report->identifier[0]->value);
+        $this->setParticipantId($report->subject->reference);
+        if (property_exists($report, 'effectiveDateTime') && $report->effectiveDateTime) {
+            $this->setDateOfDeath(new \DateTime($report->effectiveDateTime));
+        }
+        if (property_exists($report, 'valueString') && $report->valueString) {
+            $this->setCauseOfDeath($report->valueString);
+        }
+        $this->setReportMechanism($report->encounter->reference);
+        if (property_exists($report->encounter, 'display')) {
+            $this->setReportMechanismOtherDescription($report->encounter->display);
+        }
+        $this->setReportStatus($report->status);
+        foreach ($report->performer as $performer) {
+            switch ($performer->extension[0]->url) {
+                case 'https://www.pmi-ops.org/observation/authored':
+                    $this->setSubmittedBy($performer->reference);
+                    $this->setSubmittedOn(new \DateTime($performer->extension[0]->valueDateTime));
+                    break;
+                case 'https://www.pmi-ops.org/observation/reviewed':
+                    $this->setReviewedBy($performer->reference);
+                    $this->setReviewedOn(new \DateTime($performer->extension[0]->valueDateTime));
+                    break;
+            }
+
+        }
+
+        if (property_exists($report->extension[0], 'valueHumanName') && $report->extension[0]->valueHumanName) {
```

Comment:
Got a 500 error as the `$report-extension` is undefined. I think here we also need to check if the `$report->extension` is defined.

Reply:
Going to rework this so that it uses the same 'check the type rather than the index' method I did for the submitter/reviewer above.